wai-extra/Network/Wai/EventSource.hs

```diff
@@ -39,3 +41,16 @@ eventSourceAppIO src _ sendResponse =
             case eventToBuilder se of
                 Nothing -> return ()
                 Just b  -> sendChunk b >> flush >> loop
+
+-- | Make a new WAI EventSource application with a handler that emits events.
+--
+-- @since 3.0.28
+eventStreamAppRaw :: ((ServerEvent -> IO()) -> IO ()) -> Application
+eventStreamAppRaw handler _ sendResponse =
+  sendResponse $ responseStream
+      status200
+      [(hContentType, "text/event-stream")]
+      $ \sendChunk flush -> handler $ \event ->
+           case eventToBuilder event of
+               Nothing -> return ()
+               Just b  -> sendChunk b >> flush
```

Comment:
It may be more general to provide two different callback action: one to `sendChunk`, and the other to `flush`. This would mirror other APIs we already have.

Reply:
good point, will amend